beacon_node/beacon_chain/tests/block_verification.rs

```diff
@@ -1373,3 +1343,102 @@ async fn add_altair_block_to_base_chain() {
         }
     ));
 }
+
+#[tokio::test]
+async fn import_duplicate_block_unrealized_justification() {
+    let spec = MainnetEthSpec::default_spec();
+
+    let harness = BeaconChainHarness::builder(MainnetEthSpec)
+        .spec(spec)
+        .keypairs(KEYPAIRS[..].to_vec())
+        .fresh_ephemeral_store()
+        .mock_execution_layer()
+        .build();
+    let chain = &harness.chain;
+
+    // Move out of the genesis slot.
+    harness.advance_slot();
+
+    // Build the chain out to the first justification opportunity 2/3rds of the way through epoch 2.
+    let num_slots = E::slots_per_epoch() as usize * 8 / 3;
+    harness
+        .extend_chain(
+            num_slots,
+            BlockStrategy::OnCanonicalHead,
+            AttestationStrategy::AllValidators,
+        )
+        .await;
+
+    // Move into the next empty slot.
+    harness.advance_slot();
+
+    // The store's justified checkpoint must still be at epoch 0, while unrealized justification
+    // must be at epoch 1.
+    // FIXME(sproul): this also kinda seems like a bug, I don't think we should be computing
```

Comment:
Just checking that you want this `FIXME` to be merged.

Reply:
After discussion offline we think this is just a harmless spec quirk. The state is in epoch 2 at this point, so it will consider justification and finalization for epochs 1 and 2. It is legit to justify epoch 1 during epoch 2, either through unrealized justification or realized justification (if epoch 2 doesn't get enough votes).